News/Services/AiIntegrationService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.Extensions.Options;
+using News.GptModels;
+using News.Infrastructure;
+using News.Services.ServicesInterface;
+using Newtonsoft.Json;
+using System.Text;
+
+namespace News.Services;
+
+public class AiIntegrationService : IAiIntegrationService
+{
+    private readonly OpenAiOptions _options;
+    private readonly HttpClient _client;
+
+    public AiIntegrationService(IHttpClientFactory httpClientFactory, IOptions<OpenAiOptions> options)
+    {
+        _options = options.Value;
+        _client = httpClientFactory.CreateClient();
+        _client.DefaultRequestHeaders.Add("Authorization", $"Bearer {_options.ApiKey}");
+    }
+
```

Comment:
I dont remember Avan's comment btw you need to move Http client to HttpRawRequest.

Reply:
you can check the comments. They are there. 
I have implemented changes based on those comments. 
If something more is needed, I will ask you to continue writing, as I won’t have time for the next couple of days due to work